src/server/server_family.cc

```diff
@@ -446,22 +543,17 @@ void ServerFamily::Init(util::AcceptServer* acceptor, util::ListenerInterface* m
   stats_caching_task_ =
       pb_task_->AwaitBrief([&] { return pb_task_->AddPeriodic(period_ms, cache_cb); });
 
-  fs::path data_folder = fs::current_path();
-  const auto& dir = GetFlag(FLAGS_dir);
-
-  error_code file_ec;
-  if (!dir.empty()) {
-    data_folder = fs::canonical(dir, file_ec);
+  string flag_dir = GetFlag(FLAGS_dir);
+  if (IsCloudPath(flag_dir)) {
+    aws_.reset(new cloud::AWS("s3"));
```

Comment:
It's safer to use `std::make_unique<cloud::AWS>("s3")` instead of `.reset()`. Generally, calling `reset()` and `unique_ptr`'s c'tor directly is considered bad practice.